[PGNCCL] Simplify support macro definition

kwen2501 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #145964
* #145893

- Promotes usage of `NCCL_VERSION_CODE >= NCCL_VERSION(X, Y, Z)`

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

kwen2501 committed:
[PGNCCL] Simplify support macro definition

[ghstack-poisoned]

oraluben commented:
> Also bumped minimum NCCL version to 2.10

Is there any special reason to do this? 2.7 is still working now.

kwen2501 commented:
Sure, I reverted the bump.

kwen2501 committed:
Update on "[PGNCCL] Simplify support macro definition"


- Promotes usage of `NCCL_VERSION_CODE >= NCCL_VERSION(X, Y, Z)`

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj reviewed (approved):
nice!

shuqiangzhang commented on torch/csrc/distributed/c10d/NCCLUtils.hpp:
```diff
@@ -18,71 +18,43 @@
 
-#if defined(NCCL_MAJOR) && (NCCL_MAJOR == 2) && defined(NCCL_MINOR) && \
-    (NCCL_MINOR >= 14)
+#if NCCL_VERSION_CODE >= NCCL_VERSION(2, 14, 0)
```
Do you know which version of NCCL started to define the MARCROs of like NCCL_VERSION_CODE, NCCL_VERSION?

kwen2501 replied:
very very long time ago :)
